### ISSUE TITLE
Create segments for multiday events for displayed out of scope days

### ISSUE THREAD

**Issue by calin-marian** [antoniandre/vue-cal-v4#206]
I have some multi day events that spread across 2 months, e.g.: start 31 dec 2019, end 3 jan 2020.

Currently, the segments are calculated only for days in the current month, but since the change that displayes events also on out of scope days, the code looks for segments for those days too, and an error is triggered: TypeError: "event.segments[cell.formattedDate] is undefined". This is because the code is looking for an segment for 1st of Jan 2020 and 2nd of Jan 2020, but the event only has segments created for 31st of Dev 2019, and none for 1st or 2nd of Jan 2020, even though the cells are present in the view, and rendered.

The change I propose is to calculate segments for all the displyed days, so instead of: createEventSegments(e, startDate, endDate, this) we run: createEventSegments(e, firstCellDate, lastCellDate, this).

With this change, I no longer have the error, and the events are displayed correctly.

In my project I have only the month view, so I don't know if this issue is reproducible on other views, but I don't think so, because I think only on month view you have days displayed that are not in current scope (to show full week at the begining and end).

**Comment by calin-marian** [comment]
Pull request: https://github.com/antoniandre/vue-cal/pull/205

**Comment by antoniandre** [comment]
Hi @calin-marian, thanks for reporting and for the PR. 
But I need to re-check in the code first where the problem comes from, because for performance, we should not create segments we don't use.
In your case creating segments for 42 days (6 weeks x 7 days) instead of 4 days.

Will let you know soon what I find out.

**Comment by antoniandre** [comment]
This is great @calin-marian, 
My apologies for misunderstanding you at first, and taking so long to get into this issue and in the end just accepting your change!

This is totally the way to go as I already calculate the minimum amount of segments to create for a range inside the `createEventSegments` function:

```js
  let timestamp = Math.max(viewStartDate.getTime(), eventStart)
  const end = Math.min(viewEndDate.getTime(), eventEnd)
```

I am adding some small changes in that PR, will be merged shortly.

Thanks for your help!

**Comment by antoniandre** [comment]
@calin-marian,
Now published in version 2.14.1.
Demo here: https://codepen.io/antoniandre/pen/gOOJebP?editors=1010
